packages/iTwinUI-react/src/core/Table/hooks/useResizeColumns.tsx

```diff
@@ -268,24 +264,45 @@ const reducer = <T extends Record<string, unknown>>(
       nextHeaderIdWidths = [],
     } = newState.columnResizing;
 
+    if (!instance) {
+      return newState;
+    }
+
     const deltaX = clientX - startX;
 
     const newColumnWidths = getColumnWidths(
       headerIdWidths,
       deltaX / columnWidth,
     );
-    const newNextColumnWidths = getColumnWidths(
-      nextHeaderIdWidths,
-      -deltaX / nextColumnWidth,
-    );
+
+    const isTableWidthDecreasing =
+      calculateTableWidth(newColumnWidths, instance.flatHeaders) <
+      instance.tableWidth;
+    const newNextColumnWidths =
+      instance?.resizeMode === 'fit' ||
+      (instance?.resizeMode === 'expand' && isTableWidthDecreasing)
+        ? getColumnWidths(nextHeaderIdWidths, -deltaX / nextColumnWidth)
+        : {};
 
     if (
-      !isNewColumnWidthsValid(newColumnWidths, instance?.flatHeaders) ||
-      !isNewColumnWidthsValid(newNextColumnWidths, instance?.flatHeaders)
+      !isNewColumnWidthsValid(newColumnWidths, instance.flatHeaders) ||
+      !isNewColumnWidthsValid(newNextColumnWidths, instance.flatHeaders) ||
+      !isNewTableWidthValid(
+        { ...newColumnWidths, ...newNextColumnWidths },
```

Comment:
I think you are expanding next column. Maybe this could be changed to next avaible for resize? Because right now in storybook when I try to make 'Description' smaller - nothing happens. I think 'End date' should be expanded.

Reply:
'End Date` has max width so I can't expand past it. And I don't really want to write some more code to make things even more complex to check whether it reached max width and if so, expand another column. At least until we get another issue regarding this.